haystack/document_stores/opensearch.py

```diff
@@ -659,17 +792,56 @@ def _validate_approximate_knn_settings(
                 self._assert_embedding_param(
                     name="ef_search", actual=embedding_field_ef_search, expected=512, index_id=index_id
                 )
-        if self.index_type == "hnsw":
+
+        elif self.index_type == "hnsw":
+            expected_ef_construction = self.knn_parameters.get("ef_construction", 80)
+            expected_m = self.knn_parameters.get("m", 64)
+            expected_ef_search = self.knn_parameters.get("ef_search", 20)
             self._assert_embedding_param(
                 name="method.name", actual=embedding_field_method_name, expected="hnsw", index_id=index_id
             )
             self._assert_embedding_param(
-                name="ef_construction", actual=embedding_field_ef_construction, expected=80, index_id=index_id
+                name="ef_construction",
+                actual=embedding_field_ef_construction,
+                expected=expected_ef_construction,
+                index_id=index_id,
             )
-            self._assert_embedding_param(name="m", actual=embedding_field_m, expected=64, index_id=index_id)
+            self._assert_embedding_param(name="m", actual=embedding_field_m, expected=expected_m, index_id=index_id)
             if self.knn_engine == "faiss":
                 self._assert_embedding_param(
-                    name="ef_search", actual=embedding_field_ef_search, expected=20, index_id=index_id
+                    name="ef_search", actual=embedding_field_ef_search, expected=expected_ef_search, index_id=index_id
+                )
+
+    def _validate_ivf_settings(
+        self, existing_embedding_field: Dict[str, Any], index_settings: Dict[str, Any], index_id: str
+    ):
+        # Index is not trained yet and should therefore be an HNSW index with default settings until index is trained
+        if "model_id" not in existing_embedding_field:
+            self._validate_hnsw_settings(existing_embedding_field, index_settings, index_id)
```

Comment:
This doesn't do anything effectively, right? index_type is neither "flat" nor "hnsw". So I guess we can safely remove it.